### PR TITLE
Enhancement/593 object handling

### PR DESCRIPTION
**Pull request by DnlLrssn**
**Checklist**

- [ ] tests added
- [ ] commits conform to the [Commit message format](./CONTRIBUTING.md#commit)
- [ ] documentation updated

**Squash commit message**

* Add `make verify` Makefile command (executes quickbuild, test and lint-min)
* Move uniqueInts to helper package
* Move getting object functions and setting up get data events from changesheet to session
* Add functionality to register special handling of object type
* Add auto-chart handling as special handling
* Get Select paths from new objecthandler

### CODE REVIEW

File: helpers/uniqueInts_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		t.Fatalf("uniqueInts array was nil")
+	}
+
+	validateIntArraySortable(t, "uniqueInts", a, []int{5, 22}, true)
+
+	if !m.HasValue(5) {
+		t.Error("uniqueInts collection doesn't contain expected value: 5")
+	}
+
+	if m.HasValue(10) {
```
Comment (SeBBBe) [inline helpers/uniqueInts_test.go:30]: that seems.. specific?

File: session/objecthandling.go [hunk 2]
```diff
@@ -0,0 +1,741 @@
...
+		getObject := func(ctx context.Context) error {
+			var err error
+			genObj, err = sense.CurrentApp.Doc.GetObject(ctx, name)
+			return err
+		}
+		if err := sessionState.SendRequest(actionState, getObject); err != nil {
+			return errors.Wrapf(err, "Failed go get object<%s>", name)
+		}
+
+		obj, err := sense.AddNewObject(genObj.Handle, enigmahandlers.ObjTypeSheetObject, name, genObj)
```
Comment (SeBBBe) [inline session/objecthandling.go:90]: I know it's old but maybe the name "ObjTypeSheetObject" isn't ideal

File: session/objecthandling.go [hunk 3]
```diff
@@ -0,0 +1,741 @@
...
+		rawGeneratedProp, errDataPath := generatedPropPath.Lookup(rawAutoChartProperties)
+		if errDataPath != nil {
+			return errors.Wrapf(errDataPath, "Failed to get generated properties for autochart<%s>", autochartGen.GenericId)
+		}
+
+		// Create sessionObject of type
+		var genObj *enigma.GenericObject
+		createSessionObject := func(ctx context.Context) error {
+			var err error
+			genObj, err = uplink.CurrentApp.Doc.CreateSessionObjectRaw(ctx, rawGeneratedProp)
```
Comment (SeBBBe) [inline session/objecthandling.go:177]: Nice! This is much better than before

File: session/objecthandling.go [hunk 4]
```diff
@@ -0,0 +1,741 @@
...
+		} else {
+			isStackedDims = 0
+		}
+
+		datapages, err := gob.GetHyperCubeReducedData(ctx, requestDef.Path, []*enigma.NxPage{
+			{
+				Left:   isStackedDims,
+				Top:    0,
+				Width:  len(hypercube.DimensionInfo) + len(hypercube.MeasureInfo) + isStackedDims,
+				Height: int(math.Min(2000.0, 10000.0/float64(len(hypercube.MeasureInfo)+1))),
```
Comment (SeBBBe) [inline session/objecthandling.go:407]: Some magic numbers here. Could do with an explanation/constants.

File: helpers/uniqueInts_test.go (hunk 1 above)
Comment (DnlLrssn) [reply helpers/uniqueInts_test.go:30]: old file that was moved, I think it's just a test to make sure it doesn't always return true.

Comment (DnlLrssn) [reply helpers/uniqueInts_test.go:30]: i.e. value doesn't matter

File: session/objecthandling.go (hunk 2 above)
Comment (DnlLrssn) [reply session/objecthandling.go:90]: Agree, didn't do this type of refactoring. Maybe `ObjTypeGenericObject` works better?

Comment (SeBBBe) [reply session/objecthandling.go]: Yeah I think that better describes it

**Pull request merged**